### DISCUSSION TITLE
KTX import workflow

### DISCUSSION THREAD

**Discussion by atteneder** [atteneder/KtxUnity#56] [Ideas]
KtxUnity 1.2 and 2.1 added initial support for importing KTX 2.0 files in the Editor (at design-time). In this discussion I'd like to collect ideas how such a workflow should ideally look like and what features are missing. Feel free to share your requirements and use-cases.

- Option to not transcode, but keep the compressed KTX for runtime transcoding
- Option to select texture format (per-platform; similar to TextureImporter)
- Option to flip texture

Feel free to add to it

**Comment by hybridherbst** [comment]
A long-standing issue is that everything powering the TextureInspector is internal (e.g. the per-platform dropdowns, some of the compression options, sizing, ...). It's still possible to use them (e.g. via AsmRef or Reflection) but not as clean as it could be.

![image](https://user-images.githubusercontent.com/2693840/163419281-5040444e-275f-4059-a464-2f289b13975f.png)
(you could probably ask for an InternalsVisibleTo attribute though to make it cleaner👀)

Without these options, ktx2 textures dropped into Unity and used in a build might not be in the right format (e.g. they're decompressed as X but the runtime platform would have needed Y).
What would also be interesting is if this can completely be inferred from the currently active platform - e.g. the importer would unpack + keep the right files per-platform that are also used at runtime.
